### ISSUE TITLE
Arduino esp client does not receive subscribed messages

### ISSUE THREAD

**Issue by Msegun** [eclipse-arrowhead/core-java-spring#131]
Hello 

We are running coresystems in local cloud. 

Also we have two rest clients (one of them is on arduino esp32). They both register in service registry and subscribe to their own events. Then both of them are manually authorized to receive messages from each other
Rest A <- does receive published message no problem.
Rest B (the one on arduino esp) <- does not receive it (Perhaps we did something wrong during authorization)

Publish message returns status 200 so the message was sent to all authorized messages.
(It would be really useful to send 201 if at least one subscriber received the message.)

We want to debug the problem ourselves but we do not really know how to enable debug logs in spring (We did not found the way to enable debug flag like in the previous version)  so that we can see publish info in logs)

**Comment by MaGaMeGa** [comment]
Hello Msegun,
To enable debug log of Event Handler,  you have to change the log level in log4j2.xml at  ...\core-java-spring\eventhandler\src\main\resources as :
  old :   
```
<Property name="JDBC_LEVEL">INFO</Property>
```
 new:   
```
<Property name="JDBC_LEVEL">DEBUG</Property> '   
```
and/or

  old:    
```
<Property name="CONSOLE_FILE_LEVEL">INFO</Property>
```
new: 
```
<Property name="CONSOLE_FILE_LEVEL">DEBUG</Property>
```
If you change the JDBC_LEVEL property, the new log will show up in the database. If you change the CONSOLE_FILE_LEVEL property, the new log will show up in the console and in the log file.   

⚠️ 
 Even if all your authorization configuration correct, you still may run into a similar phenomenon if your subscriber ( Rest B (the one on arduino esp)  ) has a non-public ip address and it is not in the same ip subnetwork as the Event Handler. In this case, the subscriber can register to events ( it sees the Event Handler) but could not receive messages as it is not seen by the Event Handler.

**Comment by MaGaMeGa** [comment]
After changing the log level properties in log4j2.xml at  ...\core-java-spring\eventhandler\src\main\resources
for changes to take effect you will have to rebuild the project.

An alternative solution, to start a core system with a different log level without a rebuild, is to copy the changed log4j2.xml next to the jar file (same folder/directory),
and start the core system as 
```
 java -Dlog4j.configurationFile=file:log4j2.xml -jar arrowhead-eventhandler-4.1.3.jar
```

**Comment by MaGaMeGa** [comment]
When you run your client system on another device then the CoreSystem runs on,  you have to make sure, that the generated certificate for your client contains the actual IP address in its Subject Alternative Name extension. [https://github.com/arrowhead-f/core-java-spring/blob/master/documentation/certificates/create_client_certificate.pdf](url)  >> STEP 10 - add other IP addresses

**Comment by MaGaMeGa** [comment]
Please, let me know if any of the above solved your problem.
If not, please send me the logs and the corresponding service registry and authorization entries, so I could assist with the debugging process.

**Comment by Msegun** [comment]
Thanks for the quick response 

We enabled the debug it helped a lot thanks 

The problem was the way we tried to authorize our systems.
We did everything properly but in the wrong order 
What we did 
1) Register to sr
2) Registered subscriber
3) Authorized It manually (with swagger)

We noticed that fileld "sources" in subscriber /GET response was empty.

After unsubscribing and subscribing again the field contained all the necessary sources and the rest B receiver published message. 

Basically what we should've done was 
1) Register to sr
2) Authorized It 
2) Registered subscriber

Thanks for the support and quick response.

**Comment by MaGaMeGa** [comment]
Thank you for the update on the issue.

I updated the client readme with recommended Subscriber start-up order [at](https://github.com/arrowhead-f/client-skeleton-java-spring/blob/development/README.md)
as :
#### Recommended Subscriber start-up order
* Register Subscriber system in Service Registry (through swagger)
* Authorize Subscriber to Publisher in Authorization (through swagger)
* Start Subscriber (subscribe to events)
